src/Analyzer.Cli.UnitTests/CommandLineParserTests.cs

```diff
@@ -0,0 +1,98 @@
+        [DataRow("\\CommandLineParserTests.cs", 3, "--report-format", "Sarif", DisplayName = "Path exists, Report-format flag set, --output-file-path flag not included.")]
+        [DataRow("\\CommandLineParserTests.cs", 1, "--parameters-file-path", DisplayName = "Path exists, Parameters-file-path flag included, but no value provided.")]
+        [DataRow("\\AppServicesLogs-Failures.json", 5, DisplayName = "Issues found in the template")]
+        [DataRow("\\AppServicesLogs-Passes.json", 0, DisplayName = "Success")]
+        public void AnalyzeTemplate_ValidInputValues_ReturnExitCode(string relativeTemplatePath, int expectedExitCode, params string[] additionalParams)
+        {
+            var args = new string[] { "analyze-template" , String.Concat(Directory.GetCurrentDirectory(), relativeTemplatePath)}; 
+            args = args.Concat(additionalParams).ToArray();
+            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
+
+            Assert.AreEqual(expectedExitCode, result.Result);
+        }
+
+        [DataTestMethod]
+        [DataRow("\\Configuration.json", 1, DisplayName = "Provided parameters file is not a parameters file")]
+        [DataRow("\\Parameters.json", 5, DisplayName = "Provided parameters file correct, issues in template")]
+        public void AnalyzeTemplate_ParameterFileParamUsed_ReturnExitCode(string relativeParametersFilePath, int expectedExitCode)
+        {
+            var templatePath = String.Concat(Directory.GetCurrentDirectory(), "\\AppServicesLogs-Failures.json");
+            var parametersFilePath = String.Concat(Directory.GetCurrentDirectory(), relativeParametersFilePath);
+            var args = new string[] { "analyze-template", templatePath, "--parameters-file-path", parametersFilePath };
+            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
+
+            Assert.AreEqual(expectedExitCode, result.Result);
+        }
+
+        [TestMethod]
+        public void AnalyzeTemplate_ConfigurationFileParamUsed_ReturnExitCode()
+        {
+            var templatePath = String.Concat(Directory.GetCurrentDirectory(), "\\AppServicesLogs-Failures.json");
+            var configurationPath = String.Concat(Directory.GetCurrentDirectory(), "\\Configuration.json");
+            var args = new string[] { "analyze-template", templatePath, "--config-file-path", configurationPath};
+            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
+
+            Assert.AreEqual(5, result.Result);
+        }
+
+        [TestMethod]
+        public void AnalyzeTemplate_SarifRun_ReturnExitCode()
+        {
+            var templatePath = String.Concat(Directory.GetCurrentDirectory(), "\\AppServicesLogs-Failures.json");
+            var outputFilePath = String.Concat(Directory.GetCurrentDirectory(), "\\outPutFile.sarif");
+            // The FileStream has to be closed for the cli to be able to write values into it
+            using (var fs = File.Create(outputFilePath))
```

Comment:
Why do you need to create the file first?

Reply:
Because a file needs to be passed in with `Sarif` flag. I thought it would be weird to pass in an empty file just for this test to work.

Reply:
The file path with the Sarif flag just tells the CLI where to write the Sarif data to.  It doesn't mean the file needs to exist first.  If it did, then users would have to create an empty file before they could run the command, which would be a poor experience.

Reply:
Thank you for explaining.